language/extensions/move-table-extension/src/lib.rs

```diff
@@ -80,6 +82,49 @@ pub trait TableResolver {
     ) -> InternalGasUnits<GasCarrier>;
 }
 
+struct UnsafeStaticTableResolver {
```

Comment:
you probably want to add 50 lines of comments explaining the safety contract of this thing and how to use it properly

Reply:
Yeah I will want to. When I think people will agree on this solution..